File: Enigma-Machine-master/src/me/matt/enigma/wrappers/Reflector.java
package me.matt.enigma.wrappers;

public class Reflector {

    char[][] reflector;

    public Reflector(final char[][] reflector) {
        this.reflector = reflector;
    }

    public char swap(final char c) {
        for (int i = 0; i < reflector[0].length; i++) {
            if (c == reflector[0][i]) {
                return reflector[1][i];
            } else if (c == reflector[1][i]) {
                return reflector[0][i];
            }
        }
        return c;// This line should never be reached, acts as a failsafe.
    }

}

File: app/src/main/java/com/sdr/enigmasimulator/enigma/Reflector.kt
package com.sdr.enigmasimulator.enigma

class Reflector(var reflector: Array<CharArray>) {
    fun swap(c: Char): Char {
        for (i in reflector[0].indices) {
            if (c == reflector[0][i]) {
                return reflector[1][i]
            } else if (c == reflector[1][i]) {
                return reflector[0][i]
            }
        }
        return c // This line should never be reached, acts as a failsafe.
    }

}